### PR TITLE
Make sure all internal lists do not rely on keywords order

### PR DESCRIPTION
**Pull request by mvorisek**
as requested in https://github.com/doctrine/sql-formatter/pull/107#pullrequestreview-2072371411

Into 1.4.x as this is needed for #106 bugfix where the lists are massively updated based on exact SQL specs.

We keep the lists alpha sorted for easier maintenance but resort them by length in the tokenizer for longest match.

The lists should not be defined sorted by length, as with comming bugfix PR, the sorting by length will no longer be needed, as there is nothing like "multi-word reserved keyword".

### CODE REVIEW

**Comment by mvorisek** [comment]
@greg0ire @goetas @derrabus @SenseException Guys, please approve, so I can continue with follow up bugfixes.

**Review by goetas — DISMISSED** [review]
Not a fan of it but is ok from my pow

**Review by greg0ire — CHANGES_REQUESTED** [review]
There are 2 mixed concerns in this PR, one of them wasn't in the original PR. There should at least be 2 commits, each with meaningful commit messages that don't just tell us what you did, but why you did it. If you want things to go fast, I suggest you create several PRs, one for each commit, with good commit messages and explanations, rather than pinging us in the middle of the night, only one hour after creating your PR. And by "suggest", I of course mean "require".

**Comment by mvorisek** [comment]
@greg0ire PR splitted into two commits.

File: tests/TokenizerTest.php [hunk 1]
```diff
@@ -7,9 +7,44 @@
...
+    private function getTokenizerList(string $propertyName): array
+    {
+        $tokenizerReflClass = new ReflectionClass(Tokenizer::class);
+        /** @var list<string> $res */
+        $res = $tokenizerReflClass->getProperty($propertyName)->getDefaultValue();
+
+        return $res;
+    }
+
+    public function testInternalKeywordListsAreSortedForEasierMaintenance(): void
```
Comment (goetas) [inline tests/TokenizerTest.php:30]: i think that this should be a code style check rather than an unit test.

Comment (mvorisek) [reply tests/TokenizerTest.php:30]: If you know how, then great and definitely tell me, if not, let's move with this PR. In #106 I need to add many keywords and I want to make sure the git diff is minimal for now and long term.

**Comment by mvorisek** [comment]
@greg0ire I have reworked the commits to be more readable. I hope the commits answer all questions. Please merge it to 1.4.x as part of #106 bugfix.

**Review by greg0ire — APPROVED** [review]

**Pull request merged**